src/utils/gitExtension.ts

```diff
@@ -0,0 +1,33 @@
+import * as vscode from 'vscode';
+import * as semver from 'semver';
+import {API, GitExtension, Ref, Repository} from './git';
+
+function getGitExtensionAPI(): API {
+   return (<GitExtension>(
+      vscode.extensions.getExtension('vscode.git')!.exports
+   )).getAPI(1);
```

Comment:
should we handle the api 1 not being available or the extension not being available?

Reply:
yes good catch i've updated it